app/src/main/java/com/github/swent/echo/compose/authentication/CreateProfile.kt

```diff
@@ -376,3 +430,210 @@ fun InputChipFun(
+                if (rawPicture == null) {
+                    Log.w("photo picker", "cannot read the file")
+                } else {
+                    showPictureDialog = true
+                }
+            }
+        }
+    val pictureDisplaySize = 100.dp
+    val pictureStartPadding = 5.dp
+    val pictureAlpha = 0.5f
+    val deleteButtonOffset = 10.dp
+    Column(modifier = Modifier.padding(start = pictureStartPadding)) {
+        Box {
+            Image(
+                modifier =
+                    Modifier.size(pictureDisplaySize)
+                        .clip(CircleShape)
+                        .clickable {
+                            pickPhotoActivity.launch(
+                                PickVisualMediaRequest(
+                                    ActivityResultContracts.PickVisualMedia.ImageOnly
+                                )
+                            )
+                        }
+                        .testTag("profile-picture-image"),
+                painter =
+                    if (picture != null) {
+                        BitmapPainter(picture!!.asImageBitmap())
+                    } else {
+                        painterResource(R.drawable.echologoround)
+                    },
+                contentDescription = "",
+                alpha = pictureAlpha
+            )
+            Icon(
+                imageVector = Icons.Outlined.Edit,
+                contentDescription = stringResource(R.string.profile_creation_edit_picture),
+                modifier = Modifier.align(Alignment.Center)
+            )
+        }
+        IconButton(
+            modifier =
+                Modifier.align(Alignment.End)
+                    .offset(x = deleteButtonOffset, y = -deleteButtonOffset)
+                    .testTag("profile-picture-delete"),
+            onClick = { onPictureChange(null) }
+        ) {
+            Icon(
+                imageVector = Icons.Outlined.Delete,
+                contentDescription = stringResource(R.string.profile_creation_delete_picture),
+            )
+        }
+    }
+
+    if (showPictureDialog) {
+        Dialog(onDismissRequest = { showPictureDialog = false }) {
+            PictureTransformer(
+                picture = rawPicture!!,
+                onCancel = { showPictureDialog = false },
+                onConfirm = { newPicture ->
+                    showPictureDialog = false
+                    onPictureChange(newPicture)
+                }
+            )
+        }
+    }
+}
+
+/**
+ * Picture edition dialog which allow the user to center the picture
+ *
+ * @param picture: the picture to center
+ * @param onConfirm: callback called when the picture is centered
+ * @param onCancel: callback when the action is canceled
+ */
+@Composable
+fun PictureTransformer(
+    picture: Bitmap,
+    onConfirm: (picture: Bitmap) -> Unit,
+    onCancel: () -> Unit
+) {
+    val circleRadius = min(picture.width, picture.height) * 3 / 8
```

Comment:
I don't think it's a good idea to set the circle radius according the image size. I tried it with different image, and some mystical things happens if the image is not the same size as the screen. I think it would be preferable to use a fix size

Reply:
The problem here is that if the picture is smaller than the fixed size then the app crashes